raiden/tests/integration/test_raidenservice.py

```diff
@@ -348,3 +357,59 @@ def get_channel_state(app) -> NettingChannelState:
     ]
 
     assert channel_state.fee_schedule.imbalance_penalty == full_imbalance_penalty
+
+
+@raise_on_failure
+@pytest.mark.parametrize("number_of_nodes", [2])
+@pytest.mark.parametrize("channels_per_node", [0])
+@pytest.mark.parametrize("number_of_tokens", [1])
+def test_blockchain_event_processed_interleaved(
+    raiden_network: List[RaidenService],
+    token_addresses: List[TokenAddress],
+    restart_node: RestartNode,
+):
+    """ Blockchain events must be transformed into state changes and processed by
+    the state machine interleaved.
+
+    Otherwise problems arise when the creation of the state change is dependent
+    on the state of the state machine.
+
+    Regression test for: https://github.com/raiden-network/raiden/issues/6444
+    """
+    app0, app1 = raiden_network
+
+    app1.stop()
+
+    api0 = RaidenAPI(app0)
+    channel_id = api0.channel_open(
+        registry_address=app0.default_registry.address,
+        token_address=token_addresses[0],
+        partner_address=app1.address,
+    )
+    api0.channel_close(
+        registry_address=app0.default_registry.address,
+        token_address=token_addresses[0],
+        partner_address=app1.address,
+    )
+
+    # Restart node 1
+    restart_node(app1)
+    wait_all_apps(raiden_network)
+
+    # Check correct events
+    assert app1.wal, "app1.wal not set"
+    app1_state_changes = app1.wal.storage.get_statechanges_by_range(RANGE_ALL_STATE_CHANGES)
+
+    assert search_for_item(
+        app1_state_changes, ContractReceiveChannelNew, {"channel_identifier": channel_id}
+    )
+    assert not search_for_item(
+        app1_state_changes, ContractReceiveRouteNew, {"channel_identifier": channel_id}
+    )
```

Comment:
Why could we get a `ContractReceiveRouteNew` here?

Reply:
Good point,doesn't make much sense here.